test/sample-test/run_test.sh

```diff
@@ -101,242 +103,236 @@ while [ "$1" != "" ]; do
+# Utility function to prepare and validate the results for sample test based on
+# ipynb.
+# Arguments:
+#   Test name.
+################################################################################
+check_notebook_result() {
+  jupyter nbconvert --to python $1.ipynb
+  pip3 install tensorflow==1.8.0
+  ipython $1.py
+  EXIT_CODE=$?
+  cd ${TEST_DIR}
+  python3 check_notebook_results.py --experiment "$1-test$" --testname \
+  $1 --result ${SAMPLE_TEST_RESULT} --namespace ${NAMESPACE} \
+  --exit-code ${EXIT_CODE}
+
+  echo "Copy the test results to GCS ${RESULTS_GCS_DIR}/"
+  gsutil cp ${SAMPLE_TEST_RESULT} ${RESULTS_GCS_DIR}/${SAMPLE_TEST_RESULT}
+  # Need to make sure CMLE models are cleaned inside the notebook.
+}
+
+if [[ -z "$RESULTS_GCS_DIR" ]]; then
+  usage
+  exit 1
 fi
 
 if [[ ! -z "${GOOGLE_APPLICATION_CREDENTIALS}" ]]; then
-  gcloud auth activate-service-account --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
+  gcloud auth activate-service-account \
+  --key-file="${GOOGLE_APPLICATION_CREDENTIALS}"
 fi
 
-GITHUB_REPO=kubeflow/pipelines
-BASE_DIR=/python/src/github.com/${GITHUB_REPO}
-TEST_DIR=${BASE_DIR}/test/sample-test
-
 cd ${BASE_DIR}
 
 # Install argo
 echo "install argo"
 ARGO_VERSION=v2.3.0
 mkdir -p ~/bin/
 export PATH=~/bin/:$PATH
-curl -sSL -o ~/bin/argo https://github.com/argoproj/argo/releases/download/$ARGO_VERSION/argo-linux-amd64
+curl -sSL -o ~/bin/argo "https://github.com/argoproj/argo/releases/download/\
+$ARGO_VERSION/argo-linux-amd64"
 chmod +x ~/bin/argo
 
 echo "Run the sample tests..."
 
 # Run the tests
-if [ "$TEST_NAME" == 'tf-training' ]; then
-  SAMPLE_KUBEFLOW_TEST_RESULT=junit_SampleKubeflowOutput.xml
-  SAMPLE_KUBEFLOW_TEST_OUTPUT=${RESULTS_GCS_DIR}
-
-  #TODO: convert the sed commands to sed -e 's|gcr.io/ml-pipeline/|gcr.io/ml-pipeline-test/' and tag replacement. 
-  # Compile samples
-  cd ${BASE_DIR}/samples/core/kubeflow-tf
-
-  if [ -n "${DATAFLOW_TFT_IMAGE}" ];then
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_TFT_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-kubeflow-tf-trainer:\([a-zA-Z0-9_.-]\)\+|${KUBEFLOW_DNNTRAINER_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tf-predict:\([a-zA-Z0-9_.-]\)\+|${DATAFLOW_PREDICT_IMAGE}|g" kubeflow-training-classification.py
-    sed -i -e "s|gcr.io/ml-pipeline/ml-pipeline-local-confusion-matrix:\([a-zA-Z0-9_.-]\)\+|${LOCAL_CONFUSIONMATRIX_IMAGE}|g" kubeflow-training-classification.py
+preparation ${TEST_NAME}
+
+if [[ "${TEST_NAME}" == "kubeflow-training-classification" ]]; then
+  #TODO: convert the sed commands to sed -e 's|gcr.io/ml-pipeline/|gcr.io/
+  # ml-pipeline-test/' and tag replacement.
+  # Compile samples.
+
+  if [[ -n "${DATAFLOW_TFT_IMAGE}" ]]; then
+    sed -i "s|gcr.io/ml-pipeline/ml-pipeline-dataflow-tft:\([a-zA-Z0-9_.-]\)\+\
```

Comment:
It seems to me that the previous formatting looked more readable.

Reply:
Agree. Changed it back. Also breaklines tend to cause problems.